scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/RunningTaskData.java

```diff
@@ -44,7 +44,11 @@
 
     private final NodeSet nodes;
 
-    private int pingAttempts = 0;
+    private volatile int pingAttempts = 0;
```

Comment:
According to this: https://stackoverflow.com/questions/7805192/is-a-volatile-int-in-java-thread-safe, volatile int won't be thread safe if you do increment. so maybe it is better to add **sycnhronized**  to method where `pingAttempts++`

Reply:
AtomicInteger ?

Reply:
AtomicInteger is even better